### ISSUE TITLE
Predicted coordinates are only in the center

### ISSUE THREAD

**Issue by manfred-seiwald** [QihuangZhang/CeLEry#6]
**Problem** 
I used CeLEry for deconvolution of the cytospace dataset, but found that the predicted coordinates stay in the center of the reference coordinates (see image).
**Explanation**
The way how I calculated the predicted spots is:
```
def compute_spot_coords(self, pred_coords: np.ndarray, spot_locations: np.ndarray)->np.ndarray:
    # Row = x, Col = y
    xmin = spot_locations[:,0].min()
    xmax = spot_locations[:,0].max()
    ymin = spot_locations[:,1].min()
    ymax = spot_locations[:,1].max()
    x_spot_coords = (pred_coords[:,0] * (xmax - xmin) + xmin).astype(np.int32)
    y_spot_coords = (pred_coords[:,1] * (ymax - ymin) + ymin).astype(np.int32)
    spot_coords = np.stack([x_spot_coords, y_spot_coords], 1)
    return spot_coords
```
where the prediction is done by
```
spot_locations = spatial_rna.obs[['Row', 'Col']]
# create coordinate model from spatial_rna
celery.Fit_cord(data_train=spatial_rna, location_data=spot_locations, hidden_dims = [30, 25, 15],  num_epochs_max = 20, path = results_dir + '/temp', filename = 'coord_model')
# predict coordinates with model for each reference cell in ref_scrna, result is array (num_cells, 2), the coordinates are normalized 0..1
pred_coords = celery.Predict_cord(data_test=ref_scrna, path = results_dir + '/temp', filename = 'coord_model')
```
**Question**
Could you think of a reason why most of the predicted locations are centered and not well distributed ?
blue = spot_locations
orange = pred_coords
![image](https://github.com/QihuangZhang/CeLEry/assets/6967323/320cd2d7-1501-4c1e-855d-b703dd63b296)

**Comment by QihuangZhang** [comment]
Looks like the model is very underfitted. Could you try tuning the maximum epoch to be higher (20 might be too small)? Also, you could explore other hyperparameters that are related to the modeling fitting. We add a small section in the help document to illustrate their usage:

https://github.com/QihuangZhang/CeLEry/blob/main/tutorial/tutorial.md#training-tips--tuning-matters-a-lot

**Comment by manfred-seiwald** [comment]
Thank you very much for answering. I will make the test in the following days and give you a response.

**Comment by manfred-seiwald** [comment]
I have tested several variants, here the results:
- with higher initial_learning_rate (0.1, 0.01, 0.001) the loss was higher the results worse. So 0.0001 (default value) was the best initial_learning_rate with loss=0.98
- I tested with 50 epochs
- I tested with number_error_try = 30
- There was an improvement with batch_size=32, where the loss decreased: loss = 0.084, but the final result (see image) only changed marginally.

If you want I can provide you with the data, they are taken from Cytospace: the spatial file has 452 MB, the reference file has 4.4 GB. 

![pred_spots_lr_0 0001_batch_32](https://github.com/QihuangZhang/CeLEry/assets/6967323/0620d7de-6972-4edb-b20d-c1423fd0fab9)
